radix-engine/src/engine/call_frame.rs

```diff
@@ -565,6 +587,30 @@ where
 
         Ok((output, moving_buckets, moving_proofs, moving_vaults))
     }
+
+    pub fn take_cost_unit_counter(&mut self) -> CostUnitCounter {
+        self.cost_unit_counter
+            .take()
+            .expect("Frame doesn't own a cost unit counter")
+    }
+
+    pub fn cost_unit_counter(&mut self) -> &mut CostUnitCounter {
```

Comment:
I'm finding this type of method signature becoming problematic as these don't capture partial borrows of `self`.

Reply:
https://users.rust-lang.org/t/how-to-partially-borrow-from-struct/32221/2

Reply:
Wow, there is a workaround for partial borrow across functions. I will give it try.